Update OTLP guide with what's new in 3.2 (#2574)
prometheus/docs: content/docs/guides/opentelemetry.md

  translation_strategy: NoUTF8EscapingWithSuffixes
```

## Delta Temporality

> The [OpenTelemetry specification says](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#temporality) that both Delta temporality and Cumulative temporality are supported. While Delta temporality is common in systems like statsd and graphite, cumulative temporality is the default temporality for Prometheus.

Reviewer: [Nit] Delta is capitalized in this sentence while cumulative is not:
```suggestion
The [OpenTelemetry specification says](https://opentelemetry.io/docs/specs/otel/metrics/data-model/#temporality) that both Delta temporality and Cumulative temporality are supported. While delta temporality is common in systems like statsd and graphite, cumulative temporality is the default temporality for Prometheus.
```

> Today, Prometheus embeds the [deltatocumulative processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/deltatocumulativeprocessor) from [OpenTelemetry-Collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib), which is capable of ingesting deltas and transforms it into their equivalent cumulative value before they get stored in Prometheus' TSDB.

Reviewer: [Nit] We might as well write the name of the processor the same as the project itself does:
```suggestion
Today, Prometheus embeds the [delta to cumulative processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/deltatocumulativeprocessor) from [OpenTelemetry-Collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib), which is capable of ingesting deltas and transforming them into the equivalent cumulative representation before storing in Prometheus' TSDB.
```
